Makefile

```diff
@@ -18,6 +18,7 @@ dependencies:
 	go get -u golang.org/x/tools/cmd/cover
 	go get -u github.com/NebulousLabs/merkletree
 	go get -u github.com/boltdb/bolt
+	go get -u github.com/inconshreveable/muxado
```

Comment:
muxado is a stream multiplexing library. It allows us to layer multiple RPCs over the same persistent connection. It provides nice abstractions that make this feel very similar to the old one-conn-per-RPC scheme.